auth0/src/main/java/com/auth0/android/authentication/storage/CredentialsManager.java

```diff
@@ -97,6 +97,30 @@ public void onFailure(AuthenticationException error) {
         });
     }
 
+    /**
+     * Checks if a valid pair of credentials can be obtained from this manager.
+     */
+    public boolean hasCredentials() {
+        String accessToken = storage.retrieveString(KEY_ACCESS_TOKEN);
+        String refreshToken = storage.retrieveString(KEY_REFRESH_TOKEN);
+        String idToken = storage.retrieveString(KEY_ID_TOKEN);
+        Long expiresAt = storage.retrieveLong(KEY_EXPIRES_AT);
+
+        return !(isEmpty(accessToken) && isEmpty(idToken) || expiresAt == null || expiresAt <= getCurrentTimeInMillis() && refreshToken == null);
```

Comment:
parenthesis is not needed because the `||` separates the conditions. I will add them anyway to make it easier to read. 👍